WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -339,23 +340,17 @@ private void addCommentApproveActionForCommentNotification(NotificationCompat.Bu
         // adding comment approve action
         Intent commentApproveIntent = getCommentActionIntent();
         commentApproveIntent.addCategory(KEY_CATEGORY_COMMENT_MODERATE);
-        commentApproveIntent.putExtra(NotificationsListFragment.NOTE_INSTANT_APPROVE_EXTRA, true);
+        commentApproveIntent.putExtra(NotificationsProcessingService.ARG_ACTION_TYPE, NotificationsProcessingService.ARG_ACTION_APPROVE);
         if (noteId != null) {
-            commentApproveIntent.putExtra(NotificationsListFragment.NOTE_ID_EXTRA, noteId);
+            commentApproveIntent.putExtra(NotificationsProcessingService.ARG_NOTE_ID, noteId);
         }
-        PendingIntent commentApprovePendingIntent = PendingIntent.getActivity(this, 0, commentApproveIntent,
-                PendingIntent.FLAG_CANCEL_CURRENT);
+        PendingIntent commentApprovePendingIntent =  PendingIntent.getService(this, 0, commentApproveIntent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
It has been a long time since I wrote a Android Service, so my memory could be a little rusty on the topic. I've a couple of questions about starting a service with the `FLAG_UPDATE_CURRENT`.

Are we sure that `FLAG_UPDATE_CURRENT` does't give problems in case of multiple distinct notifications on the status bar?

Similar to the question above, but when the user taps Like twice in a row. EX: The user taps Like on the 1st comment, the service is started its networking, and in the meanwhile a new PN for a new comment arrives on the device. The user taps "like" on this 2nd notification... 

I think, if we want to handle multiple moderation/liking cases, we should set the the `requestCode` when getting the intent, so we can 2 or more intents at the same time. `int uniqueCodeID = new Random().nextInt(9999)` or `int uniqueCodeID = noteId` and then we can use `FLAG_UPDATE_CURRENT`. Request code could also be used later to show UI updates to the end users.